### PR TITLE
Document TLS and crypto settings

### PR DESCRIPTION
**Pull request by tiran**
Documentation is work-in-progress.

Trello card: https://trello.com/c/a9egV9TF/62-1-sccfsi-publicly-document-the-key-configurations-and-crypto-levels-in-openshifts-default-configuration

Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

**Comment by enj** [comment]
cc @openshift/security @openshift/sig-security

File: architecture/index.adoc [hunk 1]
```diff
@@ -146,3 +147,85 @@ automatically delivered into the container. This enables the infrastructure to
...
 endif::[]
+
+Communication channels with the REST API as well as between
+xref:infrastructure_components/kubernetes_infrastructure.adoc#master-components[master components]
+such as etcd and API server, are secured with
+link:https://en.wikipedia.org/wiki/Transport_Layer_Security[TLS]. It is
+perhaps better known under its former name "SSL" or by its use in "HTTPS". TLS
+provides strong encryption, data integrity, and authentication of servers with
+X.509 server certificates and public key infrastructure.
+ifdef::openshift-origin,openshift-enterprise,openshift-dedicated[]
```
Comment (php-coder) [inline architecture/index.adoc]: Who is excluded from this list? openshift-online?

Comment (tiran) [inline architecture/index.adoc]: Yes, the custom PKI detail is not relevant for openshift-online and ``../install_config`` is not built for it either.

**Comment by tiran** [comment]
The documentation is going to need another update once I have dealt with https://github.com/openshift/origin/issues/15671 . At the moment some clients use Golang defaults. I like to change that.

File: architecture/index.adoc (hunk 1 above)
Comment (adellape) [inline architecture/index.adoc]: `../install_config` isn't built for `openshift-dedicated` either.

**Comment by adellape** [comment]
@tiran Other than the `openshift-dedicated` comment, LGTM. What versions of OCP is this PR relevant for?

> The documentation is going to need another update once I have dealt with openshift/origin#15671 . At the moment some clients use Golang defaults. I like to change that.

Are you saying you want to get this merged now and follow-up later after openshift/origin#15671 is addressed?

**Comment by adellape** [comment]
[Document: [SCCFSI] publicly document the key configurations and crypto levels in OpenShift's default configuration](https://trello.com/c/GMpGBmHB/599-document-sccfsi-publicly-document-the-key-configurations-and-crypto-levels-in-openshifts-default-configuration)

**Pull request merged**